library/argument/src/ap_option_with_parameter.e

```diff
@@ -93,9 +95,13 @@ feature -- Access
 	parameter: detachable G
 			-- Last value give to the option
 		require
-			was_found: was_found
+			was_found: was_found or has_default_value
 		do
-			Result := parameters.last
+			if parameters.is_empty then
```

Comment:
It would be safer to use a local variable for `parameters', nothing prevents `parameters' to return a different object on each call. Even if in the descendants, `parameters' is usually an attribute.